### PR TITLE
Add skeleton for Game Lobby Menu

### PR DESCRIPTION
**Pull request by nuruvilu**
The lobby is opened by clicking "new game" in the main menu. Currently only two start dates are in the listbox, but the intention is for saved games to be present as well. Most of the space on the right-hand panel is reserved for information about the selected country and other players. The purpose of the separation is to allow a map to be rendered behind the menu.
The start button can be pressed only after a save has been selected.

Save games/start dates don't actually get loaded yet, since obviously the actual underlying game data doesn't exist yet.

image

### CODE REVIEW

File: game/src/LobbyMenu/LobbyMenu.gd [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	get_tree().change_scene_to_file("res://src/SampleGame.tscn")
+
+
+func _on_game_select_list_item_selected(index):
+	print("Selected save game: ", index)
+	save_game_selected.emit(index)
+
+
+func _on_save_game_selected(_index):
+	$GameStartPanel/VBoxContainer/StartButton.disabled = false
```
Comment (Spartan322) [inline game/src/LobbyMenu/LobbyMenu.gd:27]: Prefer an export variable to hardcoded node paths. Example:
```gdscript
@export
var start_button : BaseButton
# ...
start_button.disabled = false
```
This makes scene refactoring trivial.

File: game/src/LobbyMenu/LobbyMenu.gd [hunk 2]
```diff
@@ -0,0 +1,27 @@
+extends Control
+
+signal back_button_pressed
+signal save_game_selected
+
+# Called when the node enters the scene tree for the first time.
+func _ready():
+	pass # Replace with function body.
+
```
Comment (Spartan322) [inline game/src/LobbyMenu/LobbyMenu.gd:9]: Since its not used:
```suggestion
```

File: game/src/LobbyMenu/LobbyMenu.gd [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+func _on_back_button_button_down():
+	print("Returning to Main Menu.")
+	back_button_pressed.emit()
+
+
+func _on_start_button_button_down():
+	print("Starting new game.")
+	get_tree().change_scene_to_file("res://src/SampleGame.tscn")
```
Comment (Spartan322) [inline game/src/LobbyMenu/LobbyMenu.gd:18]: Prefer an export variable to a hardcoded scene file path. Example:
```gdscript
@export
var game_scene : PackedScene
# ...
get_tree().change_scene_to_packed(game_scene)
```

**Review by Spartan322 — CHANGES_REQUESTED** [review]
Seems functionally good, few things need to be changed, also would prefer to have dev cycle requirements as comments, if you're unsure I'll see about dealing with it myself in a later PR anyway.

File: game/src/LobbyMenu/LobbyMenu.gd (hunk 3 above)
Comment (nuruvilu) [reply game/src/LobbyMenu/LobbyMenu.gd:24]: For some reason using an export variable here caused the game to get stuck on the load screen, so I've left it out for now.

**Comment by nuruvilu** [comment]
Added requirements tags.

File: game/src/LobbyMenu/LobbyMenu.gd (hunk 3 above)
Comment (Spartan322) [reply game/src/LobbyMenu/LobbyMenu.gd:24]: I believe this is a case of godotengine/godot#73983, a regression introduced in Godot 4 RC 4, should be fixed in Godot 4 RC 6, I'll test it beforehand to make sure.

**Review by Spartan322 — APPROVED** [review]
Looks good.

**Pull request merged**